tools/buildsys/src/manifest.rs

```diff
@@ -167,100 +175,111 @@ kernel-parameters = [
    "console=ttyS42",
 ]
 
-`grub-features` is a list of supported grub features.
-This list allows us to conditionally use or exclude certain grub features in specific variants.
-The only supported value at this time is `set-private-var`.
-This value means that the grub config for the current variant includes the command to find the
-BOTTLEROCKET_PRIVATE partition and set the appropriate `$private` variable for the grub to
-consume.
-Adding this value to `grub-features` enables the use of Boot Config.
+`image-features` is a map of image feature flags, which can be enabled or disabled. This allows us
```

Comment:
In buildsys you have added a loop to add these as --build-args when calling docker build. However in `Dockerfile` I think you need to say `ARG GRUB_SET_PRIVATE_VAR`. In other words, these are not exactly free-form, there is an enumerable list of valid strings/keys right? Should they be constrained by enum or otherwise?

Also I don't quite understand how it goes from being `grub-set-private-var` to `GRUB_SET_PRIVATE_VAR`, I didn't see that conversion to screaming snake case.